### ISSUE TITLE
Support for Ruby 2.4.1 Absent Operator

### ISSUE THREAD

**Issue by backus** [ammar/regexp_parser#33]
See [Absent Operator](https://medium.com/rubyinside/the-new-absent-operator-in-ruby-s-regular-expressions-7c3ef6cd0b99#.5ez9id4kv).

Basically it matches anything not within the parens:

```ruby
>> "John Doe" =~ /\A(?~John) Doe\z/
=> nil
>> "Jane Doe" =~ /\A(?~John) Doe\z/
=> 0
```

Kind of weird they released this in a patch version.

**Comment by ammar** [comment]
Excellent. Thanks.

Also need to update Unicode properties to Unicode 8.0, which I believe was added to ruby in 2.4.0. It looks like it only adds new script names though.

**Comment by ammar** [comment]
Addressed in release 0.4.4
